crates/net/network/src/peers/manager.rs

```diff
@@ -283,16 +283,23 @@ impl PeersManager {
         self.connection_info.decr_pending_in();
         self.connection_info.inc_in();
 
+        // check weather if the peer is trustable or not
+        let mut is_trusted = self.trusted_peer_ids.contains(&peer_id);
+        if self.trusted_nodes_only && !is_trusted {
+            self.queued_actions.push_back(PeerAction::DisconnectUntrustedIncoming { peer_id });
+            return
```

Comment:
here we're returning after increasing the `self.connection_info.inc_in();` but we haven't updated the peer's state so it won't get reset on disconnect.
we can only inc_in after this check